hal/halx86/minihal/CMakeLists.txt

```diff
@@ -4,39 +4,11 @@ list(APPEND MINI_HAL_SOURCE
     ../legacy/bus/isabus.c
     ../legacy/bus/pcibus.c
     ../legacy/bussupp.c
-    ../generic/dma.c
-    ../generic/drive.c
-    ../generic/misc.c
-    ../generic/nmi.c
     ../generic/spinlock.c
-    ../generic/timer.c
-    ../generic/usage.c
     ../pic/pic.c
-    ../include/hal.h
-    halinit.c)
+    ../include/hal.h)
 
-if(SARCH STREQUAL "xbox")
-    list(APPEND MINI_HAL_SOURCE
-        ../generic/beep.c
-        ../generic/cmos.c
-        ../xbox/clock.c
-        ../xbox/reboot.c)
-elseif(SARCH STREQUAL "pc98")
-    list(APPEND MINI_HAL_SOURCE
-        ../pc98/beep.c
-        ../pc98/clock.c
-        ../pc98/cmos.c
-        ../pc98/reboot.c)
-else()
-    list(APPEND MINI_HAL_SOURCE
-        ../generic/beep.c
-        ../generic/clock.c
-        ../generic/cmos.c
-        ../generic/reboot.c)
```

Comment:
Not so fast. It would be of interest to actually use some of that code in freeldr, instead of reimplementing it in freeldr (i'm talking about the beep, reboot code, amongst others).

Reply:
> beep, reboot code

This is usually done by BIOS interrupts or boot services.